app/javascript/application.js

```diff
@@ -48,6 +48,18 @@ import 'jquery-ui/themes/base/sortable.css'
 import { I18n } from "i18n-js";
 import locales from "../../tmp/locales.json";
 
+// create a promise that only gets resolved when turbolinks:load fired and all select2 locales have been loaded
+let select2_locales_loaded = new Promise((resolve, reject) => {
+  document.addEventListener('turbolinks:load', resolve);
+});
+Object.keys(locales).forEach(locale =>
+    select2_locales_loaded = Promise.all([select2_locales_loaded, import(`select2/dist/js/i18n/${locale}`)])
+);
+window.select2_locales_loaded = select2_locales_loaded;
```

Comment:
While I preferred the previous version (with emitting an event) a little bit more, the following question arised: 

Is there any specific reason you chose the `window` here (and in all other places), rather than the `document`? I would probably go with `document`, that shouldn't change much technically, just semantically:

```
document.dispatchEvent(new Event("select2_locales_loaded"))
```

And, I also like the colon `:` notation, but that's just taste.

Reply:
No, there is no specific reason I used window so I switched to document. Also I am not sure what you mean by colon notation but I am guessing you mean "select2:locales:loaded" instead of "select2_locales_loaded"?

Reply:
Indeed, thanks!